app/src/main/java/de/westnordost/streetcomplete/quests/barrier_type/AddStileType.kt

```diff
@@ -60,25 +59,17 @@ class AddStileType : OsmElementQuestType<StileTypeAnswer> {
                     oldType != null && oldType != newType ||
                     newMaterial != null && oldMaterial != null && oldMaterial != newMaterial
 
+                // => properties that refer to the old replaced stile should be removed
                 if (stileWasRebuilt) {
-                    // => properties that refer to the old replaced stile should be removed
-                    tags.deleteIfExistList(STILE_PROPERTIES - "material")
-                    if(newMaterial != null) {
-                        tags["material"] = newMaterial
-                    } else {
-                        tags.remove("material")
-                    }
```

Comment:
Cool, I was going to say this looked like it could be simplified. :partying_face: